### PR TITLE
Feat: alter needReinstall param of TestTask to pass reinstallation of uploaded app file

### PR DESCRIPTION
**Pull request by olivershen-wow**
**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Code compiles correctly with all tests are passed.
- [X] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [X] No

## Pull Request Description

**A few words to explain your changes**:  
This change is related to the demand that the pass-in app installation can be passed when users want to test the pre-installed app.

Hydra Lab can pass installation of uploaded app file.
Gradle plugin added param appVersion and altered param needUninstall. 

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*
Locally enabled center/agent, and run gradle plugin locally. Result shows everything's fine.
image

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

File: gradle_plugin/src/main/groovy/com/microsoft/hydralab/utils/HydraLabAPIClient.java [hunk 1]
```diff
@@ -1,24 +1,38 @@
...
 import org.jetbrains.annotations.Nullable;
 
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
-import static com.microsoft.hydralab.utils.CommonUtils.*;
+import static com.microsoft.hydralab.utils.CommonUtils.GSON;
```
[the remark this answers is not in the document]
Comment (olivershen-wow) [reply gradle_plugin/src/main/groovy/com/microsoft/hydralab/utils/HydraLabAPIClient.java:31]: Sure, will log an item for tracking on this, and modify it in a separate PR.

**Comment by hydraxman** [comment]
Could you share why we need to change this?

File: gradle_plugin/template/gradle.properties [hunk 2]
```diff
@@ -48,22 +48,27 @@ inspectionStrategiesStr =
...
 hydraLabAPISchema = # Optional, Hydra Lab API schema. Value: {https (Default), http}.
-appPath = # Required, path to the app file.
+appPath = # Required, path to the app file. If needReinstall = false, this value is of no use but still is required.
+appVersion = # Optional, manually passed app version. If not specified, the value will be fetched from file parser of the app by default.
 authToken = # Required, Hydra Lab user authentication token for running test.
 # Optional, the name of TEAM that the creator of authToken belongs to.
 # Value: any specified TEAM names of Hydra Lab, if not specified, will be set in Backend with creator's default TEAM name.
 teamName =
+# The name "needUninstall" is @Deprecated, use param "needReinstall" instead in the latest version.
+needReinstall/needUninstall = # Optional, flag for whether app needs to be reinstalled. Value: {true (Default), false}
```
Comment (hydraxman) [inline gradle_plugin/template/gradle.properties:64]: Could you share the reason we need to change this? Which task or deliverable is this change serves for? @olivershen-wow @zhou9584

**Review by taoran6 — APPROVED** [review]

File: gradle_plugin/template/gradle.properties (hunk 2 above)
Comment (olivershen-wow) [reply gradle_plugin/template/gradle.properties]: Reverted, and alternatively use a new param skipInstall to achieve the same goal of skipping installation.

**Review by zhou9584 — DISMISSED** [review]

**Review by zhou9584 — APPROVED** [review]

**Review by taoran6 — APPROVED** [review]

**Pull request merged**